### ISSUE TITLE
only show big units

### ISSUE THREAD

**Issue by ToBinio** [erikh/fancy-duration#2]
Is there a way to only e.g. the 2 biggest units?

so currently it prints: 

`14s 119ms 676us 900ns`

but nanoseconds do not matter in my case so it would be nice to only print seconds and milliseconds

`14s 119ms`

**Comment by erikh** [comment]
Hey, just saw this. I'll add it to the list of things I want to do for 0.9. Maybe some kind of format string involving the interval indicators, like:

```rust
let fd = FancyDuration(Duration::new(61, 0)); // 1m 1s
fd.filter("m").to_string(); // => "1m"
```

Were you thinking something like that?

**Comment by erikh** [comment]
Doing this with a `Vec<&str>` might be better. The raw api would look something like:

```rust
fd.filter(vec!["m", "s"]).to_string() // 1m 1s only
```

But this could allow for constants which could be sliced:

```rust
const ALL_TIMES: [&str; 3] = ["y", "m", "d"];

fn years_only(duration: Duration) -> String {
  duration.fancy_duration().filter(ALL_TIMES[0..0].to_vec()).to_string() // years only
}
```

This could allow for similar methods like `.ceil()` which could round up if there is additional time, like add another year here if there were months or days.

Ideas here are very welcome!

**Comment by ToBinio** [comment]
I should probably explain my problem a bit better

I am currently working on a tool similar to `cargo-update` (updating programs installed via `cargo install`)
and I want to print the time it took to update the program. currently, it looks like this:

```
Name  Before Now   Time                    
ouch  0.5.0  0.5.1 1m 39s 254ms 759us 500ns
kondo 0.7.0  0.8.0 9s 795ms 249us 300ns
```

and I would like it to look like this:

```
Name  Before Now   Time                    
ouch  0.5.0  0.5.1 1m 39s
kondo 0.7.0  0.8.0 9s 795ms
```

so it only shows important time units:

if the biggest one is an hour only show hour + minute
if minute only shows minute + seconds
if seconds only show seconds + milliseconds 

so basically an automatic filter 

many something like this:

```rust
let fd = FancyDuration(Duration::new(61, 0)); // 1m 1s
fd.top(1).to_string(); // => "1m"
```

btw, thanks for considering my request

**Comment by erikh** [comment]
Ok, sure, that seems doable. I'll probably put it in front of something a little more fine-grained however.

Thanks for using the library!

**Comment by erikh** [comment]
https://github.com/erikh/fancy-duration/commit/80d7ca3634dce8e550decea04e3f53b2d411efad Look this over and tell me what you think, please.

**Comment by erikh** [comment]
I also added features for filtering. They are in https://github.com/erikh/fancy-duration/commit/c45f7e4e012d7a8a6a87f5f894ad55957975a2a9.

I will probably release this as 0.9 when I hear from you.

**Comment by ToBinio** [comment]
Overall LGTM once again big thanks

just a small thing:
I am not sure if `round` is the perfect name since it doesn't round in the mathematical sense so 1h 45m -> 2h instead it just cuts the less important values of

**Comment by erikh** [comment]
Yeah; I guess truncate is more accurate. I'll update the API and ship out 0.9.0. Will close this ticket once done.
